ocean_lib/web3_internal/test/test_contract_handler_load.py

```diff
@@ -0,0 +1,133 @@
+    assert "DTFactory" in ContractHandler._contracts
+
+
+def test_load__name_and_address(network, example_config):
+    """Tests load() from (name, address) query."""
+    addresses = ContractHandler.get_contracts_addresses(
+        network, example_config.address_file
+    )
+    target_address = addresses["DTFactory"]
+
+    test_tuple = ("DTFactory", target_address)
+
+    assert test_tuple not in ContractHandler._contracts
+
+    ContractHandler._load("DTFactory", target_address)
+
+    assert test_tuple in ContractHandler._contracts
+
+
+@pytest.mark.skip(reason="postpone until #202 #227 fixed, then revisit in #185")
+@pytest.mark.nosetup_all  # disable call to conftest.py::setup_all()
+def test_issue185_unit(monkeypatch):
+    """For #185, unit-test the root cause method, which is load"""
+    setup_issue_185(monkeypatch)
+
+    # ensure that conftest::setup_all() was not called
+    assert ConfigProvider._config is None
+    assert Web3Provider._web3 is None
+    assert ContractHandler._contracts == dict()
+    assert ContractHandler.artifacts_path is None
+
+    # actual test. Imports only come now, to avoid setting class-level attributes
+    from ocean_lib.ocean.ocean import Ocean
+    from ocean_lib.web3_internal.wallet import Wallet
+
+    private_key = os.getenv("TEST_PRIVATE_KEY1")
+    config = {"network": os.getenv("NETWORK_URL")}
+    ocean = Ocean(config)
+
+    # Ensure it's using a path like '/home/trentmc/ocean.py/venv/artifacts'
+    assert os.path.exists(ocean._config.artifacts_path)
+    assert "venv/artifacts" in ocean._config.artifacts_path
+
+    wallet = Wallet(ocean.web3, private_key=private_key)
+    assert wallet is not None
+
+    # At this point, shouldn't have any contracts cached
+    assert ContractHandler._contracts == {}
+
+    # This is the call that causes problems in system test.
+    contract = ContractHandler._get("DataTokenTemplate", None)
+    assert contract is not None
+
+    # The first call may have caused caching. So call again:)
+    contract = ContractHandler._get("DataTokenTemplate", None)
+    assert contract is not None
+
+
+@pytest.mark.skip(reason="postpone until #202 #227 fixed, then revisit in #185")
+@pytest.mark.nosetup_all  # disable call to conftest.py::setup_all()
+def test_issue185_system(monkeypatch):
+    """A system-level test, to replicate original failure seen in #185"""
+    setup_issue_185(monkeypatch)
+
+    # actual test. Imports only come now, to avoid setting class-level attributes
+    from ocean_lib.ocean.ocean import Ocean
+    from ocean_lib.web3_internal.wallet import Wallet
+
+    private_key = os.getenv("TEST_PRIVATE_KEY1")
+    config = {"network": os.getenv("NETWORK_URL")}
+    ocean = Ocean(config)
+
+    wallet = Wallet(ocean.web3, private_key=private_key)
+
+    # this failed before the fix
+    datatoken = ocean.create_data_token("Dataset name", "dtsymbol", from_wallet=wallet)
+    assert datatoken is not None
+
+
+def setup_issue_185(monkeypatch):
```

Comment:
Can you please rename this to be more suggestive? I'm not hyped about using issue names inside the repo itself, but even if we do in comments, function names should not contain issue numbers, in my opinion.

Reply:
Point taken.

I have added a docstring to give more context in a913fa4. 

Re function names with numbers, here's my thoughts:
- For non-test code: fully agree. Don't ever have issue names in the methods there
- For test code: it depends. For simple tests without much context needed, don't have. But for more complex issues that have eg a nuanced setup, it can be very helpful: rather than having 10 lines of context in the code, it points to the issue itself. That's what I did here. 

How does that sound?